pilot/cmd/pilot-agent/main.go

```diff
@@ -160,6 +161,27 @@ var (
 	}
 )
 
+func newExecCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "exec -- COMMAND [args...]",
```

Comment:
Not sure I understand why a user would run this vs just running `command` directly?

Reply:
The command must run after the agent is ready. If you start agent in background - you have no way to know when it's ready, and the app will likely start. Also likely you need shell - which in containerless is not possible.

Reply:
Just read through the context, have the same confusion with john,  

>Also likely you need shell - which in containerless is not possible.

Why is it not possible? I understand in vm it is more likely.

Reply:
Added the part to initialize agent based on Costin's comment.

Reply:
Shell: in distroless there is no shell AFAIK. 

VMs have a shell - but there is also the use case of docker ( for example in google CloudRun or other docker exec envs ), and the image might be distroless-based.

Also there is the case for proxyless gRPC ( we support it as experimental ) - where we  want to start the app after the init is done, so the files are in place.